apps/explorer/tests/test_views.py

```diff
@@ -50,3 +50,230 @@ def test_renders_pixelset_list(self):
+            omics_area=omics_area
+        )
+        experiment_tag = 'candida'
+        experiment.tags = experiment_tag
+        experiment.save()
+
+        analysis = factories.AnalysisFactory(
+            experiments=[experiment, ]
+        )
+        analysis_tag = 'msms/time'
+        analysis.tags = analysis_tag
+        analysis.save()
+        factories.PixelSetFactory(
+            analysis=analysis
+        )
+
+        # no filter
+        response = self.client.get(self.url)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=9
+        )
+
+        # tags filter
+        tag = models.Tag.objects.get(name=experiment_tag)
+        data = {
+            'tags': [tag.id, ]
+        }
+        response = self.client.get(self.url, data)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=1
+        )
+
+        tag = models.Tag.objects.get(name=analysis_tag)
+        data = {
+            'tags': [tag.id, ]
+        }
+        response = self.client.get(self.url, data)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=1
+        )
+
+    def test_all_filters(self):
+
+        # Create 8 pixelset
+        make_development_fixtures(
+            n_pixel_sets=8,
+            n_pixels_per_set=1
+        )
+
+        # Create a special pixel set which will match all filters
+        species = factories.SpeciesFactory()
+        strain = factories.StrainFactory(species=species)
+        omics_unit_type = factories.OmicsUnitTypeFactory()
+        omics_unit = factories.OmicsUnitFactory(
+            strain=strain,
+            type=omics_unit_type,
+        )
+        omics_area = factories.OmicsAreaFactory()
+        experiment = factories.ExperimentFactory(
+            omics_area=omics_area
+        )
+        experiment_tag = 'candida'
+        experiment.tags = experiment_tag
+        experiment.save()
+        analysis = factories.AnalysisFactory(
+            experiments=[experiment, ]
+        )
+        pixel_set = factories.PixelSetFactory(
+            analysis=analysis
+        )
+        factories.PixelFactory(
+            omics_unit=omics_unit,
+            pixel_set=pixel_set,
+        )
+
+        # no filter
+        response = self.client.get(self.url)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=9
+        )
+
+        # all filters
+        tag = models.Tag.objects.get(name=experiment_tag)
+        data = {
+            'species': [species.id, ],
+            'omics_unit_types': [omics_unit_type.id, ],
+            'omics_area': [omics_area.id, ],
+            'tags': [tag.id, ],
+        }
+        response = self.client.get(self.url, data)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=1
+        )
```

Comment:
We need tests when things go wrong, like no results and bad `GET` parameter given.

Reply:
Right.